teamcity: failed test: TestChangefeedTruncateRenameDrop

The following tests appear to have failed on release-19.2 (testrace): TestChangefeedTruncateRenameDrop, TestChangefeedTruncateRenameDrop/sinkless, TestChangefeedTruncateRenameDrop/enterprise

You may want to check [for open issues](https://github.com/cockroachdb/cockroach/issues?q=is%3Aissue+is%3Aopen+TestChangefeedTruncateRenameDrop).

[#1537922](https://teamcity.cockroachdb.com/viewLog.html?buildId=1537922):


```
TestChangefeedTruncateRenameDrop/enterprise
... minutes]:
database/sql.(*DB).connectionOpener(0xc00db42a80, 0x6a76780, 0xc00accc080)
	/usr/local/go/src/database/sql/sql.go:1000 +0x140
created by database/sql.OpenDB
	/usr/local/go/src/database/sql/sql.go:670 +0x2b4

goroutine 63441 [chan receive]:
github.com/cockroachdb/cockroach/pkg/storage/closedts/provider.(*Provider).Notify.func1(0x6a76840, 0xc001a40510)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/closedts/provider/provider.go:217 +0x1ae
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker.func1(0xc006cd27e0, 0xc00abe86e0, 0xc0092790a0)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:196 +0x160
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:189 +0xc4

goroutine 63696 [sync.Cond.Wait]:
runtime.goparkunlock(...)
	/usr/local/go/src/runtime/proc.go:307
sync.runtime_notifyListWait(0xc008a37890, 0xc000025ea0)
	/usr/local/go/src/runtime/sema.go:510 +0xf9
sync.(*Cond).Wait(0xc008a37880)
	/usr/local/go/src/sync/cond.go:56 +0x8e
github.com/cockroachdb/cockroach/pkg/storage.(*raftScheduler).worker(0xc0010314d0, 0x6a76840, 0xc001a40c00)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/scheduler.go:192 +0x9c
github.com/cockroachdb/cockroach/pkg/storage.(*raftScheduler).Start.func2(0x6a76840, 0xc001a40c00)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/scheduler.go:161 +0x56
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker.func1(0xc0068ccbf0, 0xc00abe86e0, 0xc0068ccbe0)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:196 +0x160
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:189 +0xc4

goroutine 65120 [chan receive, 41 minutes]:
github.com/cockroachdb/cockroach/pkg/storage.(*Replica).RangeFeed(0xc00465a900, 0xc00c00b900, 0x6abd9c0, 0xc00c430ec0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/replica_rangefeed.go:229 +0x90d
github.com/cockroachdb/cockroach/pkg/storage.(*Store).RangeFeed(0xc00147bc00, 0xc00c00b900, 0x6abd9c0, 0xc00c430ec0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/store.go:2213 +0x4c6
github.com/cockroachdb/cockroach/pkg/storage.(*Stores).RangeFeed(0xc007531550, 0xc00c00b900, 0x6abd9c0, 0xc00c430ec0, 0x5ab6980)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/stores.go:206 +0x24b
github.com/cockroachdb/cockroach/pkg/server.(*Node).RangeFeed(0xc0089d6a00, 0xc00c00b900, 0x6abd9c0, 0xc00c430ec0, 0x6a76780, 0xc002a96300)
	/go/src/github.com/cockroachdb/cockroach/pkg/server/node.go:1033 +0x7e
github.com/cockroachdb/cockroach/pkg/rpc.internalClientAdapter.RangeFeed.func1(0xc00a2fb870, 0x6a30100, 0xc0089d6a00, 0xc00c00b900, 0x6a76780, 0xc002a96480, 0xc00c356240, 0xc00c3562a0)
	/go/src/github.com/cockroachdb/cockroach/pkg/rpc/context.go:610 +0xcc
created by github.com/cockroachdb/cockroach/pkg/rpc.internalClientAdapter.RangeFeed
	/go/src/github.com/cockroachdb/cockroach/pkg/rpc/context.go:608 +0x12e

goroutine 63745 [select, 41 minutes]:
github.com/cockroachdb/cockroach/pkg/server.periodicallyPersistHLCUpperBound(0xc00129f4f0, 0xc003923340, 0xc0068cd900, 0x5d09da8, 0xc008a8a480, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/server/server.go:1030 +0x1f8
github.com/cockroachdb/cockroach/pkg/server.(*Server).startPersistingHLCUpperBound.func2(0x6a76840, 0xc0018b5f80)
	/go/src/github.com/cockroachdb/cockroach/pkg/server/server.go:1104 +0xde
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker.func1(0xc0068cd910, 0xc00abe86e0, 0xc002910690)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:196 +0x160
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunWorker
	/go/src/github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:189 +0xc4
I191014 14:23:02.501369 1 rand.go:85  Random seed: -3946147195552034504



TestChangefeedTruncateRenameDrop
--- FAIL: testrace/TestChangefeedTruncateRenameDrop (0.000s)
Test ended in panic.




TestChangefeedTruncateRenameDrop/sinkless
...is range at key /Table/60 [r30] (zone config)
I191014 14:26:51.237618 62064 sql/event_log.go:130  [n1,client=127.0.0.1:43882,user=root] Event: "create_table", target: 60, info: {TableName:d.public.drop Statement:CREATE TABLE drop (a INT8 PRIMARY KEY) User:root}
I191014 14:26:51.954272 62064 sql/sqlbase/structured.go:1516  [n1,client=127.0.0.1:43882,user=root] publish: descID=60 (drop) version=2 mtime=1970-01-01 00:00:00 +0000 UTC
I191014 14:26:52.156068 62064 sql/event_log.go:130  [n1,client=127.0.0.1:43882,user=root] Event: "drop_table", target: 60, info: {TableName:d.public.drop Statement:DROP TABLE drop User:root CascadeDroppedViews:[]}
I191014 14:26:52.225290 62064 sql/sqlbase/structured.go:1516  [n1,client=127.0.0.1:43882,user=root,scExec] publish: descID=60 (drop) version=3 mtime=1970-01-01 00:00:00 +0000 UTC
I191014 14:26:52.749493 61341 util/stop/stopper.go:542  quiescing; tasks left:
2      [async] storage.IntentResolver: cleanup txn records
13     gossip update (merge)
10     gossip update (split)
1      [async] transport racer
1      [async] intent_resolver_ir_batcher
1      [async] intent_resolver_gc_batcher
1      [async] closedts-rangefeed-subscriber
I191014 14:26:52.749900 61341 util/stop/stopper.go:542  quiescing; tasks left:
2      [async] storage.IntentResolver: cleanup txn records
13     gossip update (merge)
10     gossip update (split)
1      [async] transport racer
1      [async] intent_resolver_ir_batcher
1      [async] closedts-rangefeed-subscriber
W191014 14:26:52.750060 63439 storage/intentresolver/intent_resolver.go:818  failed to gc transaction record: could not GC completed transaction anchored at /Table/SystemConfigSpan/Start: node unavailable; try another peer
I191014 14:26:52.755778 61528 storage/queue.go:520  [n1,s1] rate limited in gossip update (merge): node unavailable; try another peer
I191014 14:26:52.772680 61388 kv/transport_race.go:108  transport race promotion: ran 18 iterations on up to 1120 requests
I191014 14:26:52.773013 61341 util/stop/stopper.go:542  quiescing; tasks left:
5      gossip update (split)
3      gossip update (merge)
1      [async] transport racer
1      [async] closedts-rangefeed-subscriber
I191014 14:26:52.774161 61341 util/stop/stopper.go:542  quiescing; tasks left:
4      gossip update (split)
3      gossip update (merge)
1      [async] transport racer
1      [async] closedts-rangefeed-subscriber
I191014 14:26:52.774449 61341 util/stop/stopper.go:542  quiescing; tasks left:
4      gossip update (split)
2      gossip update (merge)
1      [async] transport racer
1      [async] closedts-rangefeed-subscriber
I191014 14:26:52.774709 61341 util/stop/stopper.go:542  quiescing; tasks left:
4      gossip update (split)
2      gossip update (merge)
1      [async] closedts-rangefeed-subscriber
I191014 14:26:52.774988 61341 util/stop/stopper.go:542  quiescing; tasks left:
3      gossip update (split)
2      gossip update (merge)
1      [async] closedts-rangefeed-subscriber
I191014 14:26:52.775233 61341 util/stop/stopper.go:542  quiescing; tasks left:
3      gossip update (split)
1      gossip update (merge)
1      [async] closedts-rangefeed-subscriber
I191014 14:26:52.775539 61341 util/stop/stopper.go:542  quiescing; tasks left:
2      gossip update (split)
1      gossip update (merge)
1      [async] closedts-rangefeed-subscriber
I191014 14:26:52.775849 61341 util/stop/stopper.go:542  quiescing; tasks left:
1      gossip update (split)
1      gossip update (merge)
1      [async] closedts-rangefeed-subscriber
I191014 14:26:52.776037 61528 storage/queue.go:520  [n1,s1] rate limited in gossip update (split): node unavailable; try another peer
I191014 14:26:52.776500 61341 util/stop/stopper.go:542  quiescing; tasks left:
1      gossip update (merge)
1      [async] closedts-rangefeed-subscriber
I191014 14:26:52.777106 61341 util/stop/stopper.go:542  quiescing; tasks left:
1      [async] closedts-rangefeed-subscriber




```

Please assign, take a look and update the issue accordingly.

--- a/pkg/sql/sqlbase/structured.pb.go
+++ b/pkg/sql/sqlbase/structured.pb.go
@@ -73,7 +73,7 @@ func (x *ConstraintValidity) UnmarshalJSON(data []byte) error {
 	return nil
 }
 func (ConstraintValidity) EnumDescriptor() ([]byte, []int) {
-	return fileDescriptor_structured_50daf820b4f615f7, []int{0}
+	return fileDescriptor_structured_d9b38487027f89a8, []int{0}
 }
 
 type ForeignKeyReference_Action int32
@@ -118,7 +118,7 @@ func (x *ForeignKeyReference_Action) UnmarshalJSON(data []byte) error {
 	return nil
 }
 func (ForeignKeyReference_Action) EnumDescriptor() ([]byte, []int) {
-	return fileDescriptor_structured_50daf820b4f615f7, []int{0, 0}
+	return fileDescriptor_structured_d9b38487027f89a8, []int{0, 0}
 }
 
 // Match is the algorithm used to compare composite keys.
@@ -158,7 +158,7 @@ func (x *ForeignKeyReference_Match) UnmarshalJSON(data []byte) error {
 	return nil
 }
 func (ForeignKeyReference_Match) EnumDescriptor() ([]byte, []int) {
-	return fileDescriptor_structured_50daf820b4f615f7, []int{0, 1}
+	return fileDescriptor_structured_d9b38487027f89a8, []int{0, 1}
 }
 
 // The direction of a column in the index.
@@ -195,7 +195,7 @@ func (x *IndexDescriptor_Direction) UnmarshalJSON(data []byte) error {
 	return nil
 }
 func (IndexDescriptor_Direction) EnumDescriptor() ([]byte, []int) {
-	return fileDescriptor_structured_50daf820b4f615f7, []int{6, 0}
+	return fileDescriptor_structured_d9b38487027f89a8, []int{6, 0}
 }
 
 // The type of the index.
@@ -232,7 +232,7 @@ func (x *IndexDescriptor_Type) UnmarshalJSON(data []byte) error {
 	return nil
 }
 func (IndexDescriptor_Type) EnumDescriptor() ([]byte, []int) {
-	return fileDescriptor_structured_50daf820b4f615f7, []int{6, 1}
+	return fileDescriptor_structured_d9b38487027f89a8, []int{6, 1}
 }
 
 type ConstraintToUpdate_ConstraintType int32
@@ -275,7 +275,7 @@ func (x *ConstraintToUpdate_ConstraintType) UnmarshalJSON(data []byte) error {
 	return nil
 }
 func (ConstraintToUpdate_ConstraintType) EnumDescriptor() ([]byte, []int) {
-	return fileDescriptor_structured_50daf820b4f615f7, []int{7, 0}
+	return fileDescriptor_structured_d9b38487027f89a8, []int{7, 0}
 }
 
 // A descriptor within a mutation is unavailable for reads, writes
@@ -340,7 +340,7 @@ func (x *DescriptorMutation_State) UnmarshalJSON(data []byte) error {
 	return nil
 }
 func (DescriptorMutation_State) EnumDescriptor() ([]byte, []int) {
-	return fileDescriptor_structured_50daf820b4f615f7, []int{8, 0}
+	return fileDescriptor_structured_d9b38487027f89a8, []int{8, 0}
 }
 
 // Direction of mutation.
@@ -383,7 +383,7 @@ func (x *DescriptorMutation_Direction) UnmarshalJSON(data []byte) error {
 	return nil
 }
 func (DescriptorMutation_Direction) EnumDescriptor() ([]byte, []int) {
-	return fileDescriptor_structured_50daf820b4f615f7, []int{8, 1}
+	return fileDescriptor_structured_d9b38487027f89a8, []int{8, 1}
 }
 
 // State is set if this TableDescriptor is in the process of being added or deleted.
@@ -434,7 +434,7 @@ func (x *TableDescriptor_State) UnmarshalJSON(data []byte) error {
 	return nil
 }
 func (TableDescriptor_State) EnumDescriptor() ([]byte, []int) {
-	return fileDescriptor_structured_50daf820b4f615f7, []int{9, 0}
+	return fileDescriptor_structured_d9b38487027f89a8, []int{9, 0}
 }
 
 // AuditMode indicates which auditing actions to take when this table is used.
@@ -471,7 +471,7 @@ func (x *TableDescriptor_AuditMode) UnmarshalJSON(data []byte) error {
 	return nil
 }
 func (TableDescriptor_AuditMode) EnumDescriptor() ([]byte, []int) {
-	return fileDescriptor_structured_50daf820b4f615f7, []int{9, 1}
+	return fileDescriptor_structured_d9b38487027f89a8, []int{9, 1}
 }
 
 type ForeignKeyReference struct {
@@ -493,7 +493,7 @@ func (m *ForeignKeyReference) Reset()         { *m = ForeignKeyReference{} }
 func (m *ForeignKeyReference) String() string { return proto.CompactTextString(m) }
 func (*ForeignKeyReference) ProtoMessage()    {}
 func (*ForeignKeyReference) Descriptor() ([]byte, []int) {
-	return fileDescriptor_structured_50daf820b4f615f7, []int{0}
+	return fileDescriptor_structured_d9b38487027f89a8, []int{0}
 }
 func (m *ForeignKeyReference) XXX_Unmarshal(b []byte) error {
 	return m.Unmarshal(b)
@@ -561,7 +561,7 @@ func (m *ForeignKeyConstraint) Reset()         { *m = ForeignKeyConstraint{} }
 func (m *ForeignKeyConstraint) String() string { return proto.CompactTextString(m) }
 func (*ForeignKeyConstraint) ProtoMessage()    {}
 func (*ForeignKeyConstraint) Descriptor() ([]byte, []int) {
-	return fileDescriptor_structured_50daf820b4f615f7, []int{1}
+	return fileDescriptor_structured_d9b38487027f89a8, []int{1}
 }
 func (m *ForeignKeyConstraint) XXX_Unmarshal(b []byte) error {
 	return m.Unmarshal(b)
@@ -608,7 +608,7 @@ func (m *ColumnDescriptor) Reset()         { *m = ColumnDescriptor{} }
 func (m *ColumnDescriptor) String() string { return proto.CompactTextString(m) }
 func (*ColumnDescriptor) ProtoMessage()    {}
 func (*ColumnDescriptor) Descriptor() ([]byte, []int) {
-	return fileDescriptor_structured_50daf820b4f615f7, []int{2}
+	return fileDescriptor_structured_d9b38487027f89a8, []int{2}
 }
 func (m *ColumnDescriptor) XXX_Unmarshal(b []byte) error {
 	return m.Unmarshal(b)
@@ -664,7 +664,7 @@ func (m *ColumnFamilyDescriptor) Reset()         { *m = ColumnFamilyDescriptor{}
 func (m *ColumnFamilyDescriptor) String() string { return proto.CompactTextString(m) }
 func (*ColumnFamilyDescriptor) ProtoMessage()    {}
 func (*ColumnFamilyDescriptor) Descriptor() ([]byte, []int) {
-	return fileDescriptor_structured_50daf820b4f615f7, []int{3}
+	return fileDescriptor_structured_d9b38487027f89a8, []int{3}
 }
 func (m *ColumnFamilyDescriptor) XXX_Unmarshal(b []byte) error {
 	return m.Unmarshal(b)
@@ -710,7 +710,7 @@ func (m *InterleaveDescriptor) Reset()         { *m = InterleaveDescriptor{} }
 func (m *InterleaveDescriptor) String() string { return proto.CompactTextString(m) }
 func (*InterleaveDescriptor) ProtoMessage()    {}
 func (*InterleaveDescriptor) Descriptor() ([]byte, []int) {
-	return fileDescriptor_structured_50daf820b4f615f7, []int{4}
+	return fileDescriptor_structured_d9b38487027f89a8, []int{4}
 }
 func (m *InterleaveDescriptor) XXX_Unmarshal(b []byte) error {
 	return m.Unmarshal(b)
@@ -754,7 +754,7 @@ func (m *InterleaveDescriptor_Ancestor) Reset()         { *m = InterleaveDescrip
 func (m *InterleaveDescriptor_Ancestor) String() string { return proto.CompactTextString(m) }
 func (*InterleaveDescriptor_Ancestor) ProtoMessage()    {}
 func (*InterleaveDescriptor_Ancestor) Descriptor() ([]byte, []int) {
-	return fileDescriptor_structured_50daf820b4f615f7, []int{4, 0}
+	return fileDescriptor_structured_d9b38487027f89a8, []int{4, 0}
 }
 func (m *InterleaveDescriptor_Ancestor) XXX_Unmarshal(b []byte) error {
 	return m.Unmarshal(b)
@@ -799,7 +799,7 @@ func (m *PartitioningDescriptor) Reset()         { *m = PartitioningDescriptor{}
 func (m *PartitioningDescriptor) String() string { return proto.CompactTextString(m) }
 func (*PartitioningDescriptor) ProtoMessage()    {}
 func (*PartitioningDescriptor) Descriptor() ([]byte, []int) {
-	return fileDescriptor_structured_50daf820b4f615f7, []int{5}
+	return fileDescriptor_structured_d9b38487027f89a8, []int{5}
 }
 func (m *PartitioningDescriptor) XXX_Unmarshal(b []byte) error {
 	return m.Unmarshal(b)
@@ -842,7 +842,7 @@ func (m *PartitioningDescriptor_List) Reset()         { *m = PartitioningDescrip
 func (m *PartitioningDescriptor_List) String() string { return proto.CompactTextString(m) }
 func (*PartitioningDescriptor_List) ProtoMessage()    {}
 func (*PartitioningDescriptor_List) Descriptor() ([]byte, []int) {
-	return fileDescriptor_structured_50daf820b4f615f7, []int{5, 0}
+	return fileDescriptor_structured_d9b38487027f89a8, []int{5, 0}
 }
 func (m *PartitioningDescriptor_List) XXX_Unmarshal(b []byte) error {
 	return m.Unmarshal(b)
@@ -887,7 +887,7 @@ func (m *PartitioningDescriptor_Range) Reset()         { *m = PartitioningDescri
 func (m *PartitioningDescriptor_Range) String() string { return proto.CompactTextString(m) }
 func (*PartitioningDescriptor_Range) ProtoMessage()    {}
 func (*PartitioningDescriptor_Range) Descriptor() ([]byte, []int) {
-	return fileDescriptor_structured_50daf820b4f615f7, []int{5, 1}
+	return fileDescriptor_structured_d9b38487027f89a8, []int{5, 1}
 }
 func (m *PartitioningDescriptor_Range) XXX_Unmarshal(b []byte) error {
 	return m.Unmarshal(b)
@@ -1023,7 +1023,7 @@ func (m *IndexDescriptor) Reset()         { *m = IndexDescriptor{} }
 func (m *IndexDescriptor) String() string { return proto.CompactTextString(m) }
 func (*IndexDescriptor) ProtoMessage()    {}
 func (*IndexDescriptor) Descriptor() ([]byte, []int) {
-	return fileDescriptor_structured_50daf820b4f615f7, []int{6}
+	return fileDescriptor_structured_d9b38487027f89a8, []int{6}
 }
 func (m *IndexDescriptor) XXX_Unmarshal(b []byte) error {
 	return m.Unmarshal(b)
@@ -1074,7 +1074,7 @@ func (m *ConstraintToUpdate) Reset()         { *m = ConstraintToUpdate{} }
 func (m *ConstraintToUpdate) String() string { return proto.CompactTextString(m) }
 func (*ConstraintToUpdate) ProtoMessage()    {}
 func (*ConstraintToUpdate) Descriptor() ([]byte, []int) {
-	return fileDescriptor_structured_50daf820b4f615f7, []int{7}
+	return fileDescriptor_structured_d9b38487027f89a8, []int{7}
 }
 func (m *ConstraintToUpdate) XXX_Unmarshal(b []byte) error {
 	return m.Unmarshal(b)
@@ -1128,7 +1128,7 @@ func (m *DescriptorMutation) Reset()         { *m = DescriptorMutation{} }
 func (m *DescriptorMutation) String() string { return proto.CompactTextString(m) }
 func (*DescriptorMutation) ProtoMessage()    {}
 func (*DescriptorMutation) Descriptor() ([]byte, []int) {
-	return fileDescriptor_structured_50daf820b4f615f7, []int{8}
+	return fileDescriptor_structured_d9b38487027f89a8, []int{8}
 }
 func (m *DescriptorMutation) XXX_Unmarshal(b []byte) error {
 	return m.Unmarshal(b)
@@ -1440,7 +1440,7 @@ func (m *TableDescriptor) Reset()         { *m = TableDescriptor{} }
 func (m *TableDescriptor) String() string { return proto.CompactTextString(m) }
 func (*TableDescriptor) ProtoMessage()    {}
 func (*TableDescriptor) Descriptor() ([]byte, []int) {
-	return fileDescriptor_structured_50daf820b4f615f7, []int{9}
+	return fileDescriptor_structured_d9b38487027f89a8, []int{9}
 }
 func (m *TableDescriptor) XXX_Unmarshal(b []byte) error {
 	return m.Unmarshal(b)
@@ -1732,7 +1732,7 @@ func (m *TableDescriptor_SchemaChangeLease) Reset()         { *m = TableDescript
 func (m *TableDescriptor_SchemaChangeLease) String() string { return proto.CompactTextString(m) }
 func (*TableDescriptor_SchemaChangeLease) ProtoMessage()    {}
 func (*TableDescriptor_SchemaChangeLease) Descriptor() ([]byte, []int) {
-	return fileDescriptor_structured_50daf820b4f615f7, []int{9, 0}
+	return fileDescriptor_structured_d9b38487027f89a8, []int{9, 0}
 }
 func (m *TableDescriptor_SchemaChangeLease) XXX_Unmarshal(b []byte) error {
 	return m.Unmarshal(b)
@@ -1770,7 +1770,7 @@ func (m *TableDescriptor_CheckConstraint) Reset()         { *m = TableDescriptor
 func (m *TableDescriptor_CheckConstraint) String() string { return proto.CompactTextString(m) }
 func (*TableDescriptor_CheckConstraint) ProtoMessage()    {}
 func (*TableDescriptor_CheckConstraint) Descriptor() ([]byte, []int) {
-	return fileDescriptor_structured_50daf820b4f615f7, []int{9, 1}
+	return fileDescriptor_structured_d9b38487027f89a8, []int{9, 1}
 }
 func (m *TableDescriptor_CheckConstraint) XXX_Unmarshal(b []byte) error {
 	return m.Unmarshal(b)
@@ -1873,7 +1873,7 @@ func (m *TableDescriptor_NameInfo) Reset()         { *m = TableDescriptor_NameIn
 func (m *TableDescriptor_NameInfo) String() string { return proto.CompactTextString(m) }
 func (*TableDescriptor_NameInfo) ProtoMessage()    {}
 func (*TableDescriptor_NameInfo) Descriptor() ([]byte, []int) {
-	return fileDescriptor_structured_50daf820b4f615f7, []int{9, 2}
+	return fileDescriptor_structured_d9b38487027f89a8, []int{9, 2}
 }
 func (m *TableDescriptor_NameInfo) XXX_Unmarshal(b []byte) error {
 	return m.Unmarshal(b)
@@ -1913,7 +1913,7 @@ func (m *TableDescriptor_Reference) Reset()         { *m = TableDescriptor_Refer
 func (m *TableDescriptor_Reference) String() string { return proto.CompactTextString(m) }
 func (*TableDescriptor_Reference) ProtoMessage()    {}
 func (*TableDescriptor_Reference) Descriptor() ([]byte, []int) {
-	return fileDescriptor_structured_50daf820b4f615f7, []int{9, 3}
+	return fileDescriptor_structured_d9b38487027f89a8, []int{9, 3}
 }
 func (m *TableDescriptor_Reference) XXX_Unmarshal(b []byte) error {
 	return m.Unmarshal(b)
@@ -1950,7 +1950,7 @@ func (m *TableDescriptor_MutationJob) Reset()         { *m = TableDescriptor_Mut
 func (m *TableDescriptor_MutationJob) String() string { return proto.CompactTextString(m) }
 func (*TableDescriptor_MutationJob) ProtoMessage()    {}
 func (*TableDescriptor_MutationJob) Descriptor() ([]byte, []int) {
-	return fileDescriptor_structured_50daf820b4f615f7, []int{9, 4}
+	return fileDescriptor_structured_d9b38487027f89a8, []int{9, 4}
 }
 func (m *TableDescriptor_MutationJob) XXX_Unmarshal(b []byte) error {
 	return m.Unmarshal(b)
@@ -1993,7 +1993,7 @@ func (m *TableDescriptor_SequenceOpts) Reset()         { *m = TableDescriptor_Se
 func (m *TableDescriptor_SequenceOpts) String() string { return proto.CompactTextString(m) }
 func (*TableDescriptor_SequenceOpts) ProtoMessage()    {}
 func (*TableDescriptor_SequenceOpts) Descriptor() ([]byte, []int) {
-	return fileDescriptor_structured_50daf820b4f615f7, []int{9, 5}
+	return fileDescriptor_structured_d9b38487027f89a8, []int{9, 5}
 }
 func (m *TableDescriptor_SequenceOpts) XXX_Unmarshal(b []byte) error {
 	return m.Unmarshal(b)
@@ -2033,7 +2033,7 @@ func (m *TableDescriptor_SequenceOpts_SequenceOwner) String() string {
 }
 func (*TableDescriptor_SequenceOpts_SequenceOwner) ProtoMessage() {}
 func (*TableDescriptor_SequenceOpts_SequenceOwner) Descriptor() ([]byte, []int) {
-	return fileDescriptor_structured_50daf820b4f615f7, []int{9, 5, 0}
+	return fileDescriptor_structured_d9b38487027f89a8, []int{9, 5, 0}
 }
 func (m *TableDescriptor_SequenceOpts_SequenceOwner) XXX_Unmarshal(b []byte) error {
 	return m.Unmarshal(b)
@@ -2059,7 +2059,13 @@ func (m *TableDescriptor_SequenceOpts_SequenceOwner) XXX_DiscardUnknown() {
 var xxx_messageInfo_TableDescriptor_SequenceOpts_SequenceOwner proto.InternalMessageInfo
 
 type TableDescriptor_Replacement struct {
-	ID   ID            `protobuf:"varint,1,opt,name=id,casttype=ID" json:"id"`
+	ID ID `protobuf:"varint,1,opt,name=id,casttype=ID" json:"id"`
+	// Time is just used for debugging purposes. It is not used in business
+	// logic. It is an HLC rather than just wall time only for historical
+	// reasons. Prior to 20.1 it was populated with the commit timestamp of the
+	// transaction which created this replacement. In 20.1 and after it is
+	// populated with the read timestamp at which the descriptor being
+	// replaced was read.
 	Time hlc.Timestamp `protobuf:"bytes,2,opt,name=time" json:"time"`
 }
 
@@ -2067,7 +2073,7 @@ func (m *TableDescriptor_Replacement) Reset()         { *m = TableDescriptor_Rep
 func (m *TableDescriptor_Replacement) String() string { return proto.CompactTextString(m) }
 func (*TableDescriptor_Replacement) ProtoMessage()    {}
 func (*TableDescriptor_Replacement) Descriptor() ([]byte, []int) {
-	return fileDescriptor_structured_50daf820b4f615f7, []int{9, 6}
+	return fileDescriptor_structured_d9b38487027f89a8, []int{9, 6}
 }
 func (m *TableDescriptor_Replacement) XXX_Unmarshal(b []byte) error {
 	return m.Unmarshal(b)
@@ -2104,7 +2110,7 @@ func (m *TableDescriptor_GCDescriptorMutation) Reset()         { *m = TableDescr
 func (m *TableDescriptor_GCDescriptorMutation) String() string { return proto.CompactTextString(m) }
 func (*TableDescriptor_GCDescriptorMutation) ProtoMessage()    {}
 func (*TableDescriptor_GCDescriptorMutation) Descriptor() ([]byte, []int) {
-	return fileDescriptor_structured_50daf820b4f615f7, []int{9, 7}
+	return fileDescriptor_structured_d9b38487027f89a8, []int{9, 7}
 }
 func (m *TableDescriptor_GCDescriptorMutation) XXX_Unmarshal(b []byte) error {
 	return m.Unmarshal(b)
@@ -2143,7 +2149,7 @@ func (m *DatabaseDescriptor) Reset()         { *m = DatabaseDescriptor{} }
 func (m *DatabaseDescriptor) String() string { return proto.CompactTextString(m) }
 func (*DatabaseDescriptor) ProtoMessage()    {}
 func (*DatabaseDescriptor) Descriptor() ([]byte, []int) {
-	return fileDescriptor_structured_50daf820b4f615f7, []int{10}
+	return fileDescriptor_structured_d9b38487027f89a8, []int{10}
 }
 func (m *DatabaseDescriptor) XXX_Unmarshal(b []byte) error {
 	return m.Unmarshal(b)
@@ -2201,7 +2207,7 @@ func (m *Descriptor) Reset()         { *m = Descriptor{} }
 func (m *Descriptor) String() string { return proto.CompactTextString(m) }
 func (*Descriptor) ProtoMessage()    {}
 func (*Descriptor) Descriptor() ([]byte, []int) {
-	return fileDescriptor_structured_50daf820b4f615f7, []int{11}
+	return fileDescriptor_structured_d9b38487027f89a8, []int{11}
 }
 func (m *Descriptor) XXX_Unmarshal(b []byte) error {
 	return m.Unmarshal(b)
@@ -11156,10 +11162,10 @@ var (
 )
 
 func init() {
-	proto.RegisterFile("sql/sqlbase/structured.proto", fileDescriptor_structured_50daf820b4f615f7)
+	proto.RegisterFile("sql/sqlbase/structured.proto", fileDescriptor_structured_d9b38487027f89a8)
 }
 
-var fileDescriptor_structured_50daf820b4f615f7 = []byte{
+var fileDescriptor_structured_d9b38487027f89a8 = []byte{
 	// 3473 bytes of a gzipped FileDescriptorProto
 	0x1f, 0x8b, 0x08, 0x00, 0x00, 0x00, 0x00, 0x00, 0x02, 0xff, 0xb4, 0x5a, 0xcb, 0x6f, 0x23, 0x47,
 	0x7a, 0x57, 0xf3, 0xcd, 0x8f, 0xaf, 0x66, 0x69, 0x66, 0x4c, 0xcb, 0x5e, 0x91, 0xa2, 0x3d, 0xb6,

--- a/pkg/sql/sqlbase/structured.proto
+++ b/pkg/sql/sqlbase/structured.proto
@@ -799,6 +799,12 @@ message TableDescriptor {
   message Replacement {
     option (gogoproto.equal) = true;
     optional uint32 id = 1 [(gogoproto.nullable) = false, (gogoproto.customname) = "ID", (gogoproto.casttype) = "ID"];
+    // Time is just used for debugging purposes. It is not used in business
+    // logic. It is an HLC rather than just wall time only for historical
+    // reasons. Prior to 20.1 it was populated with the commit timestamp of the
+    // transaction which created this replacement. In 20.1 and after it is
+    // populated with the read timestamp at which the descriptor being
+    // replaced was read.
     optional util.hlc.Timestamp time = 2 [(gogoproto.nullable) = false];
   }
 

--- a/pkg/sql/truncate.go
+++ b/pkg/sql/truncate.go
@@ -190,7 +190,10 @@ func (p *planner) truncateTable(
 	tableDesc.DropJobID = dropJobID
 	newTableDesc := sqlbase.NewMutableCreatedTableDescriptor(tableDesc.TableDescriptor)
 	newTableDesc.ReplacementOf = sqlbase.TableDescriptor_Replacement{
-		ID: id, Time: p.txn.CommitTimestamp(),
+		ID: id,
+		// NB: Time is just used for debugging purposes. See the comment on the
+		// field for more details.
+		Time: p.txn.ReadTimestamp(),
 	}
 	newTableDesc.SetID(0)
 	newTableDesc.Version = 1

